nnvm/src/compiler/graph_fuse.cc

```diff
@@ -161,6 +161,103 @@ nnvm::Graph GraphFusePartition(nnvm::Graph g) {
       }
     }
   }
+
+  /*
+     Above algorithm will not fuse a node whose output is fed to more than one
+     child node. This is because in general, it does not make sense to fuse multiple
+     children branches with their parent, as in the following example.
+
+            conv2d
+            /  |  \
+           /   |   \
+         op    op   op
+          |    |    |
+          |    |    |
+
+     However, when all children branches meet at a certain node, there is a possibility for
+     further operator fusion. For example, all nodes in the following subgraph can be fused
+     into a single node, if three 'in-between' nodes and the bottom node are all element wise
+     operation.
+
+            conv2d
+            /  |  \
+           /   |   \
+         op    op   op
+          \    |    /
+           \   |   /
+          elemwise add
+               |
+
+     This pattern is not uncommon. For example, it arises when conv2d op is followed by exponential
+     linear unit. If bias add and batch normalization are also present, they can be fused as well.
+
+     In fact, above fusion algorithm already fuses three in-between nodes and the element wise
+     add node in the figure above. The following code fuses the conv2d node with the already
+     fused children nodes. The following patterns are supported.
+
+     * Any number of child nodes from the top node
+     * The path from the top node to bottom node can contain any number of element wise ops.
+
+     The only restriction is that in-between nodes cannot have more than one child.
```

Comment:
What will happen if in-between nodes have parents not belonging to this block, such as skip-connection block?

Reply:
My rule only applies when all in-between children nodes and the bottom node are fused by existing fusion algorithm (see my figure above). If some in-between node has more than one parent, existing fusion algorithm won't fuse this node with other in-between nodes.  Hope this makes sense.

Just to be sure, I make sure that in-between node has only one parent, [here](https://github.com/dmlc/tvm/pull/1548/files#diff-a22b9a5a336566480a2b0be06b0c3c9cR218).

So for example, my rule does nothing to Resnet.